zetaclient/zetabridge/query_test.go

```diff
@@ -0,0 +1,872 @@
+package zetabridge
+
+import (
+	"github.com/cosmos/cosmos-sdk/client/grpc/tmservice"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	feemarkettypes "github.com/evmos/ethermint/x/feemarket/types"
+	"github.com/stretchr/testify/require"
+	tmtypes "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/zeta-chain/zetacore/cmd/zetacored/config"
+	"github.com/zeta-chain/zetacore/pkg/chains"
+	"github.com/zeta-chain/zetacore/pkg/coin"
+	crosschainTypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+	"github.com/zeta-chain/zetacore/zetaclient/interfaces"
+	"github.com/zeta-chain/zetacore/zetaclient/keys"
+	"github.com/zeta-chain/zetacore/zetaclient/metrics"
+	"github.com/zeta-chain/zetacore/zetaclient/testutils/stub"
+	"go.nhat.io/grpcmock"
+	"go.nhat.io/grpcmock/planner"
+
+	"net"
+	"testing"
+)
+
+func setupMockServer(t *testing.T, serviceFunc any, method string, input any, expectedOutput any) *grpcmock.Server {
+	l, err := net.Listen("tcp", "127.0.0.1:9090")
+	require.NoError(t, err)
+
+	server := grpcmock.MockUnstartedServer(
+		grpcmock.RegisterService(serviceFunc),
+		grpcmock.WithPlanner(planner.FirstMatch()),
+		grpcmock.WithListener(l),
+		func(s *grpcmock.Server) {
+			s.ExpectUnary(method).
+				UnlimitedTimes().
+				WithPayload(input).
+				Return(expectedOutput)
+		},
+	)(t)
+
+	return server
+}
+
+func closeMockServer(t *testing.T, server *grpcmock.Server) {
+	err := server.Close()
+	require.NoError(t, err)
+}
+
+func setupCorBridge() (*ZetaCoreBridge, error) {
+	return NewZetaCoreBridge(
+		&keys.Keys{},
+		"127.0.0.1",
+		"",
+		"zetachain_7000-1",
+		false,
+		&metrics.TelemetryServer{})
+}
+
+func TestZetaCoreBridge_GetBallot(t *testing.T) {
+	expectedOutput := observertypes.QueryBallotByIdentifierResponse{
+		BallotIdentifier: "456",
+		Voters:           nil,
+		ObservationType:  0,
+		BallotStatus:     0,
+	}
+	input := observertypes.QueryBallotByIdentifierRequest{BallotIdentifier: "123"}
+	method := "/zetachain.zetacore.observer.Query/BallotByIdentifier"
+	server := setupMockServer(t, observertypes.RegisterQueryServer, method, input, expectedOutput)
+	server.Serve()
+	defer closeMockServer(t, server)
+
+	zetabridge, err := setupCorBridge()
+	require.NoError(t, err)
+
+	resp, err := zetabridge.GetBallotByID("123")
+	require.NoError(t, err)
+	require.Equal(t, expectedOutput, *resp)
```

Comment:
Why do we expect request with `BallotIdentifier: "123"` to return a response with `BallotIdentifier: "456",`?